dlib/pixel.h

```diff
@@ -771,6 +865,158 @@ namespace dlib
             return(c2);
         }
 
+        // -----------------------------
+
+        struct Lab
+        {
+            double l;
+            double a;
+            double b;
+        };
+        /*
+            Calculate Lab from RGB
+            L is between 0 and 100
```

Comment:
Hmm, this is tough one. The L is lightness in percent, by definition the range is [0;100] [see second paragraph](https://en.wikipedia.org/wiki/Lab_color_space#CIELAB). I have chosen *signed char* so the pixel is internally consistent.

Your observation is correct though. There are significant losses in the dynamic range of L. **I will check some other implementations out there to see how do they work.** 

Scaling and shifting the L from [0;100] -> [-128;127] would be a solution, but then the actual value would not be usable, because the expected values are in [0;100].

Another option would be to switch to *float* but I'm not sure if the 4x increase in memory consumption is not overkill.